surveymonkey/surveymonkey.py

```diff
@@ -272,3 +379,148 @@ def completion(self, request, suffix=''):
             "online_help_token": "online_help_token",
         }
         return Response(LOADER.render_template("static/html/surveymonkey_completion_page.html", context))
+
+    def overwrite_survey_question_headings(self):
+        """
+        Overwrites the survey question headings.
+
+        Takes the values returned from self.get_overwritten_question_from_field
+        to overwrite the question heading with the user's response.
+
+        Raises:
```

Comment:
this shouldn't happen, you should not call this method if the data is not valid, I think this will not happen because the validate_field_data only `set overwrite_survey_questions` to true if previous_survey_id and self.survey_id were defined

Reply:
What? validate_field_data never sets overwrite_survey_questions to anything. @andrey-canon Can you explain what you mean?

Reply:
you are right the method doesn't set that, but it allows to set it if you don't add a validation error message, you are validating `previous_survey_id` and `survey_id `, that means, this method will never be called current logic if `previous_survey_id` and `survey_id are None

Reply:
But the validations must have been done already, since this method is only called in the lms view not in the studio view. @andrey-canon

Reply:
lms and studio use the same data

Reply:
So?